### ISSUE TITLE
Compatibility with composer v2

### ISSUE THREAD

**Issue by patricknelson** [clue/graph-composer#58]
Build is not compatible with composer v2 (for build).  

	$ composer build
	
	[1/1] Creating phar graph-composer-1.1.0-5-g25ea081.phar
	  - Adding main package "clue/graph-composer"
	  - Adding composer base files

	Catchable fatal error: Argument 1 passed to Clue\PharComposer\Package\Package::__construct() must be of the type array, boolean given, called in /root/graph-composer/vendor/clue/phar-composer/src/Clue/PharComposer/Phar/PharComposer.php on line 136 and defined in /root/graph-composer/vendor/clue/phar-composer/src/Clue/PharComposer/Package/Package.php on line 21
	Script @php build.php handling the build event returned with error code 255

Build still works fine however in composer v1 (now deprecated). I noticed this when attempting to containerize this (see https://github.com/patricknelson/graph-composer-docker).

**Step to reproduce:**

```bash
# Get graph-composer-docker repository and work from the 'composer-2' branch
git clone git@github.com:patricknelson/graph-composer-docker.git
cd graph-composer-docker
git checkout composer-2

# Tries to build container, but will fail
docker-compose up -d --build
```

Basically, what the `Dockerfile` does:

- Install composer 2 
- Clone down https://github.com/clue/graph-composer.git
- Installs dev dependencies
- Runs `composer build` (and fails)

**Comment by clue** [comment]
@patricknelson Thanks for reporting, I can indeed reproduce this with the current version!

For the reference: This only affects the build scripts in this project, using the package itself seems to work just fine with Composer 2 and Composer 1, so I'm marking this as a maintenance ticket.

The issue has already been fixed upstream with https://github.com/clue/phar-composer/pull/114 and we only need to update the dependency here. On top of this, installation of phar-composer is currently broken because some dependencies have been removed. This has already been fixed upstream via https://github.com/clue/phar-composer/pull/119, but there's currently no tagged release yet.

I'm currently in the process of preparing the next release of phar-composer and will update this project afterwards :shipit:
